### PR TITLE
python312Packages.{etcd3,limits}: fix build

### PR DESCRIPTION
**Pull request by Moraxyc**
add env `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION` to make protobuf compatible with old versions
See https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates

must be merge after https://github.com/NixOS/nixpkgs/commit/dcd5f162a6754db89a4a2db44094687cf97dcfac

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/etcd3/default.nix [hunk 1]
```diff
@@ -49,10 +58,10 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "etcd3" ];
 
-  meta = with lib; {
+  meta = {
     description = "Python client for the etcd API v3";
     homepage = "https://github.com/kragniz/python-etcd3";
-    license = licenses.asl20;
+    license = lib.licenses.asl20;
     maintainers = [ ];
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/etcd3/default.nix:65]: Mind become maintainer?

Comment (Moraxyc) [reply pkgs/development/python-modules/etcd3/default.nix]: Fine

**Comment by Aleksanaa** [comment]
2 packages marked as broken and skipped:
python311Packages.flask-appbuilder python311Packages.flask-appbuilder.dist

8 packages failed to build:
python311Packages.flask-limiter python311Packages.flask-limiter.dist python311Packages.libretranslate python311Packages.libretranslate.dist python311Packages.limits python311Packages.limits.dist python311Packages.slowapi python311Packages.slowapi.dist

4 packages built:
python311Packages.etcd3 python311Packages.etcd3.dist python312Packages.etcd3 python312Packages.etcd3.dist

**Comment by Aleksanaa** [comment]
Most importantly, `python3.11-aetcd-1.0.0a4.drv`

```
Checking runtime dependencies for aetcd-1.0.0a4-py3-none-any.whl
  - protobuf<5,>4 not satisfied by version 5.28.2
```

**Comment by Moraxyc** [comment]
I have tried directly replacing `protobuf>4,<5` in `aetcd` then the build succeeded, but I am not sure if this will affect the function

**Comment by Moraxyc** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.flask-appbuilder</li>
    <li>python311Packages.flask-appbuilder.dist</li>
    <li>python312Packages.flask-appbuilder</li>
    <li>python312Packages.flask-appbuilder.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 35 packages built:</summary>
  <ul>
    <li>calibre-web</li>
    <li>calibre-web.dist</li>
    <li>fittrackee</li>
    <li>fittrackee.dist</li>
    <li>frigate</li>
    <li>irrd</li>
    <li>irrd.dist</li>
    <li>libretranslate (python312Packages.libretranslate)</li>
    <li>libretranslate.dist (python312Packages.libretranslate.dist)</li>
    <li>octoprint</li>
    <li>octoprint.dist</li>
    <li>python311Packages.aetcd</li>
    <li>python311Packages.aetcd.dist</li>
    <li>python311Packages.etcd3</li>
    <li>python311Packages.etcd3.dist</li>
    <li>python311Packages.flask-limiter</li>
    <li>python311Packages.flask-limiter.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.limits</li>
    <li>python311Packages.limits.dist</li>
    <li>python311Packages.slowapi</li>
    <li>python311Packages.slowapi.dist</li>
    <li>python312Packages.aetcd</li>
    <li>python312Packages.aetcd.dist</li>
    <li>python312Packages.etcd3</li>
    <li>python312Packages.etcd3.dist</li>
    <li>python312Packages.flask-limiter</li>
    <li>python312Packages.flask-limiter.dist</li>
    <li>python312Packages.limits</li>
    <li>python312Packages.limits.dist</li>
    <li>python312Packages.slowapi</li>
    <li>python312Packages.slowapi.dist</li>
    <li>theharvester</li>
    <li>theharvester.dist</li>
  </ul>
</details>

**Comment by Moraxyc** [comment]
> I have tried directly replacing `protobuf>4,<5` in `aetcd` then the build succeeded, but I am not sure if this will affect the function

All packages that depend on `aetcd` successfully built and passed the test, I think there should be no problem

File: pkgs/development/python-modules/aetcd/default.nix [hunk 2]
```diff
@@ -28,7 +28,8 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace setup.py \
-      --replace-fail "setuptools_scm==6.3.2" "setuptools_scm"
+      --replace-fail "setuptools_scm==6.3.2" "setuptools_scm"\
+      --replace-fail "protobuf>4,<5" "protobuf"
```
Comment (fabaff) [inline pkgs/development/python-modules/aetcd/default.nix:32]: This was already fixed with https://github.com/NixOS/nixpkgs/commit/dcd5f162a6754db89a4a2db44094687cf97dcfac (currently in `staging-next`).

**Pull request merged**